src/core/net/srp_client.cpp

```diff
@@ -721,6 +730,15 @@ void Client::SendUpdate(void)
 
     VerifyOrExit(message != nullptr, error = kErrorNoBufs);
     SuccessOrExit(error = PrepareUpdateMessage(*message));
+
+    if (message->GetLength() + sizeof(Ip6::Udp::Header) + sizeof(Ip6::Header) >= Ip6::kMaxDatagramLength)
```

Comment:
Preparing the message twice seems not as efficient as it can. 
Can we add as many as possible services in `PrepareUpdateMessage` until the MTU is full?

Reply:
The main goal for this PR is addressing https://github.com/openthread/openthread/issues/7742 with a simple and small solution and avoid adding complexity and/or code-size overhead. My target was less than 0.5KB (500 bytes - and this PR is now about 264). Being efficient under this situation may not be as important. Adding as many services in message up to MTU is technically possible but will be more complex to implement.

I think the matter use-case from https://github.com/openthread/openthread/issues/7742 (adding multiple services that won't fit in an MTU) is expected be rare/unlikely situation.  If such use-case become common I think there may be a need to reconsider some design choices (either in how matter behaves and/or the use or SRP over UDP).

Reply:
SGTM to solve a simplified problem.